lib/internal/http2/util.js

```diff
@@ -415,16 +417,18 @@ function mapToHeaders(map,
       if (isArray) {
         for (var k = 0; k < value.length; k++) {
           val = String(value[k]);
-          ret.push([key, val]);
+          ret += `${key}\0${val}\0`;
         }
+        count += value.length;
       } else {
         val = String(value);
-        ret.push([key, val]);
+        ret += `${key}\0${val}\0`;
+        count++;
       }
     }
   }
 
-  return ret;
+  return [ ret, count ];
```

Comment:
IMO we should go for consistent styling here (no spaces). By the way, I would actually like to enable linting against these with `array-bracket-spacing: [error, never]` (only 7 errors in `lib/*`, but quite some churn in `test`).